api/leaderboard.go

```diff
@@ -10,842 +10,761 @@
+	if in.Body.Increment == 0 {
+		return nil, status.New(codes.InvalidArgument, "increment is required").Err()
+	}
 
-		var payload incrementScorePayload
+	lg := app.Logger.With(
+		zap.String("handler", "IncrementScore"),
+		zap.String("leaderboard", in.LeaderboardId),
+		zap.String("memberPublicID", in.MemberPublicId),
+	)
 
-		err := WithSegment("Payload", c, func() error {
-			if err := LoadJSONPayload(&payload, c, lg); err != nil {
-				app.AddError()
-				return err
-			}
-			return nil
-		})
-		if err != nil {
-			return FailWith(400, err.Error(), c)
-		}
-
-		var member *leaderboard.Member
-		err = WithSegment("Model", c, func() error {
-			lg.Debug("Incrementing member score.", zap.Int("increment", payload.Increment))
-			member, err = app.Leaderboards.IncrementMemberScore(c.StdContext(), leaderboardID, memberPublicID,
-				payload.Increment, scoreTTL)
+	var member *leaderboard.Member
+	err := withSegment("Model", ctx, func() error {
+		var err error
+		lg.Debug("Incrementing member score.", zap.Int64("increment", int64(in.Body.Increment)))
+		member, err = app.Leaderboards.IncrementMemberScore(context.Background(), in.LeaderboardId, in.MemberPublicId,
+			int(in.Body.Increment), getScoreTTL(in.ScoreTTL))
 
-			if err != nil {
-				lg.Error("Member score increment failed.", zap.Error(err))
-				app.AddError()
+		if err != nil {
+			lg.Error("Member score increment failed.", zap.Error(err))
+			app.AddError()
+			if _, ok := err.(*util.LeaderboardExpiredError); ok {
+				return status.New(codes.InvalidArgument, err.Error()).Err()
+			} else {
 				return err
 			}
-			lg.Debug("Member score increment succeeded.")
-			return nil
-		})
-		if err != nil {
-			return FailWithError(err, c)
 		}
-
-		return SucceedWith(serializeMember(member, -1, scoreTTL != ""), c)
+		lg.Debug("Member score increment succeeded.")
+		return nil
+	})
+	if err != nil {
+		return nil, err
 	}
+
+	return newDefaultMemberResponse(member), nil
 }
 
 //RemoveMemberHandler removes a member from a leaderboard
-func RemoveMemberHandler(app *App) func(c echo.Context) error {
-	return func(c echo.Context) error {
-		leaderboardID := c.Param("leaderboardID")
-		memberPublicID := c.Param("memberPublicID")
-		lg := app.Logger.With(
-			zap.String("handler", "RemoveMemberHandler"),
-			zap.String("leaderboard", leaderboardID),
-			zap.String("memberPublicID", memberPublicID),
-		)
+func (app *App) RemoveMember(ctx context.Context, in *api.RemoveMemberRequest) (*api.RemoveMemberResponse, error) {
+	lg := app.Logger.With(
+		zap.String("handler", "RemoveMember"),
+		zap.String("leaderboard", in.LeaderboardId),
+		zap.String("memberPublicID", in.MemberPublicId),
+	)
+
+	err := withSegment("Model", ctx, func() error {
+		lg.Debug("Removing member.")
+		err := app.Leaderboards.RemoveMember(ctx, in.LeaderboardId, in.MemberPublicId)
+
+		if err != nil && !strings.HasPrefix(err.Error(), notFoundError) {
```

Comment:
Even checking it first would be capable of providing not found errors because of concurrent writes. This would need to be refactored to be done in a transaction checking and then removing the member. Added to TODO.

Reply:
A "best effort" check would be OK here, imo, but we could think about changing the lib error return, then.

Thanks for the TODO. SG.